app/models/deal.rb

```diff
@@ -0,0 +1,43 @@
+class Deal < ApplicationRecord
+  PUBLISHABLE_DEAL_IMAGE_QUANTITY = 2
+  PUBLISHABLE_DEAL_QUANTITY = 10
+  DEAL_WITH_SAME_PUBLISH_DATE_QUANTITY = 2
+  DATE_DIFFERENCE = 1
+  
+  has_many_attached :images
+  has_many :line_items, dependent: :destroy
+  has_many :orders, through: :line_items, dependent: :destroy
+  belongs_to :admin, class_name: "User", foreign_key: 'created_by'
```

Comment:
let's keep this as created_by. 
deal.created_by is more descriptive than deal.admin

Reply:
Changed `admin` to `created_by`